### ISSUE TITLE
Runtime error for two classes in one ts file.

### ISSUE THREAD

**Issue by vlechenko** [Microsoft/TypeScript#8572]
**TypeScript Version:** 

1.8.10

**Code**

```ts
export class BaseLayoutModel extends BaseModel {
    constructor(attributes?: any, options?: any) {
        super(attributes, options);
    }
}

export class BaseModel extends Backbone.Model {
    constructor(attributes?: any, options?: any) {
        super(attributes, options);
    }

    destroy(options?: Backbone.ModelDestroyOptions): any {
        const opts = _.extend({
            contentType: "application/json",
            dataType: "text"
        }, options || {});
        super.destroy(opts);
    }
}
```

**Expected behavior:**

**Actual behavior:** 
Runtime error. BaseModel  is undefined

**Comment by aleksey-bykov** [comment]
in your code `BaseModel` is declared lexically after `BaseLayoutModel`

in the compiled code it means that `BaseModel` is used before being initialized

try swapping the declarations (`BaseModel` class should come first)

**Comment by vlechenko** [comment]
I know about “why it failed”. But this code translated to JS without any errors.

**Comment by Arnavion** [comment]
Duplicate of #4341 and #5207 which should be fixed by #4343
